### ISSUE TITLE
"The room was closed."

### ISSUE THREAD

**Issue by hamburgstaller** [haxball/haxball-issues#1279]
Sometimes my rooms do not appear in the "room list". Sometimes it appears again. And some people say "The room was closed." he's getting the warning. However, the room is open and I can enter (via link).

**Comment by thenorthstar** [comment]
> Sometimes my rooms do not appear in the "room list". Sometimes it appears again. And some people say "The room was closed." he's getting the warning. However, the room is open and I can enter (via link).

@yusufemirxd @basro

I'm suffering from the same problem. About 2 weeks ago I have done some updates on my bots. And the I have seen that room turns invisible when it stays inactive (no joins/leaves). In the beginning, I had thought that it might be related to what I have done, but last monday I have seen not, when I have ran another bot which uses the older scripts of that bot.

In the other hand, I have seen that the room controls (on headless console window) were also inactive for a while. That is, I couldn't use room controls (hence their autocomplete feature) for a while and it's interestingly back in life when I wait for a while on headless tab.

To sum up, I think that we have a new bug in our hand, which waits to be solved. Thanks.

And this one is bonus:

![image](https://user-images.githubusercontent.com/68077608/111026623-38a0ed80-83fc-11eb-962d-6efca6bca0c4.png)

This screen is from console window of my obstacle course bot. In fact, all the maps are changed per every 5 minutes if not enough amount of goals scored, but you can see that the maps are not changed in 5 minutes when there's inactivity. For example, the first one (with ID 22) is changing after 6 mins whereas the other one is changing 4 minutes.

**Comment by thenorthstar** [comment]
> Sometimes my rooms do not appear in the "room list". Sometimes it appears again. And some people say "The room was closed." he's getting the warning. However, the room is open and I can enter (via link).

@yusufemirxd @basro

I'm suffering from the same problem. About 2 weeks ago I have done some updates on my bots. And the I have seen that room turns invisible when it stays inactive (no joins/leaves). In the beginning, I had thought that it might be related to what I have done, but last monday I have seen not, when I have ran another bot which uses the older scripts of that bot.

In the other hand, I have seen that the room controls (on headless console window) were also inactive for a while. That is, I couldn't use room controls (hence their autocomplete feature) for a while and it's interestingly back in life when I wait for a while on headless tab.

To sum up, I think that we have a new bug in our hand, which waits to be solved. Thanks.

And this one is bonus:
![image](https://user-images.githubusercontent.com/68077608/111026747-0217a280-83fd-11eb-8c66-26842c6dd19e.png)

This screen is from console window of my obstacle course bot. In fact, all the maps are changed per every 5 minutes if not enough amount of goals scored, but you can see that the maps are not changed in 5 minutes when there's inactivity. For example, the first one (with ID 22) is changing after 6 mins whereas the other one is changing 4 minutes.

**Comment by basro** [comment]
The error message "The room was closed" is a bit misleading, it actually means the room is currently not connected to the master room server.

It's possible that your server's connection was disrupted temporarily. When trying to connect to it's room id you'll get that error message, eventually if the connection is reestablished your room will reappear and you'll be able to join it through the room link.

@thenorthstar thank you for sharing, from what you describe about the console being unresponsive it would seem the browser is freezing. it's not necessarily the same problem that @yusufemirxd is having. Please let me know if you happen to find a way to reproduce your issue.

**Comment by hamburgstaller** [comment]
> The error message "The room was closed" is a bit misleading, it actually means the room is currently not connected to the master room server.
> 
> It's possible that your server's connection was disrupted temporarily. When trying to connect to it's room id you'll get that error message, eventually if the connection is reestablished your room will reappear and you'll be able to join it through the room link.
> 
> @thenorthstar thank you for sharing, from what you describe about the console being unresponsive it would seem the browser is freezing. it's not necessarily the same problem that @yusufemirxd is having. Please let me know if you happen to find a way to reproduce your issue.

Thank you for your feedback, but I have a problem. Some can enter the room and some get this error message. Why are 2 players not getting the error message at the same time?
